src/cmd/usb.js

```diff
@@ -169,6 +169,43 @@ module.exports = class UsbCommand {
 			.then(() => console.log('Done.'));
 	}
 
+	cloudStatus(args){
+		const { until, params: { deviceID } } = args;
+		const options = { id: deviceID, api: this._api, auth: this._auth };
+		const deviceMgr = {
+			_: null,
+			set(usbDevice){
+				this._ = usbDevice || null;
+				return this;
+			},
+			close(){
+				const { _: usbDevice } = this;
+				return usbDevice ? usbDevice.close() : Promise.resolve();
+			},
+			status(){
+				return this._.getCloudConnectionStatus()
+					.then(status => status.toLowerCase());
+			}
+		};
+
+		return spin(openUsbDeviceById(options), 'Querying device...')
+			.then(usbDevice => deviceMgr.set(usbDevice))
+			.then(deviceMgr => deviceMgr.status())
+			.then(status => {
+				if (until && until !== status){
+					throw new Error(`Unexpected status: ${status}`);
+				}
+				console.log(status);
+			})
+			.catch(error => {
+				if (until){
+					return this.cloudStatus(args);
```

Comment:
🤪going to need a default timeout too

Reply:
✅ in https://github.com/particle-iot/particle-cli/pull/577/commits/8977b22cfb8f1973681402dbcbc308dbc24342b4